### DISCUSSION TITLE
how to enable and use precompiled header for header only packages in vcpkg

### DISCUSSION THREAD

**Discussion by niyue** [microsoft/vcpkg#22849] [Q&A]
CMake previously introduced `target_precompile_headers` [1] in CMake 3.16, I wonder if there is any support in vcpkg about this feature so that vcpkg users can take advantage of it. 

I would like to know if there is any switch in vcpkg that can enable a port, in particular those header only ports, to precompile its headers, and refer those precompiled headers in a consistent way. Or should this support be added to a port one by one by modifying their port definitions?

I did some web search previously but found nothing relevant. Could any one shed some light? Thanks.

[1] CMake's `target_precompile_headers`, https://cmake.org/cmake/help/git-stage/command/target_precompile_headers.html

**Comment by Osyotr** [comment]
`target_precompile_headers` has no relation to vcpkg whatsoever. If you need precompiled headers, use them - you don't need anything else.
